vpnglean/CMakeLists.txt

```diff
@@ -6,28 +6,70 @@ add_library(vpnglean STATIC IMPORTED GLOBAL)
 
 set_target_properties(vpnglean PROPERTIES FOLDER "Libs")
 
+set(RUST_ARCH "")
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
+    set(RUST_ARCH "x86_64-apple-darwin")
+elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
+    if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "ARM64")
+        set(RUST_ARCH "aarch64-pc-windows-msvc")
+    elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86")
+        set(RUST_ARCH "i686-pc-windows-msvc")
+    elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "AMD64")
+        set(RUST_ARCH "x86_64-pc-windows-msvc")
+    endif()
+elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+    set(RUST_ARCH "x86_64-unknown-linux-gnu")
+elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
+    if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "aarch64")
+        set(RUST_ARCH "aarch64-linux-android")
+        set(OPENSSL_ARCH "arm64")
+    elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "armv7-a")
+        set(RUST_ARCH "armv7-linux-androideabi")
+        set(OPENSSL_ARCH "arm")
+    elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "i686")
+        set(RUST_ARCH "i686-linux-android")
+        set(OPENSSL_ARCH "x86")
+    elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64")
+        set(RUST_ARCH "x86_64-linux-android")
+        set(OPENSSL_ARCH "x86_64")
+    endif()
+endif()
+
+if (RUST_ARCH STREQUAL "")
+    message(FATAL_ERROR "Unable to determine Rust target for ${CMAKE_SYSTEM_NAME} (${CMAKE_SYSTEM_PROCESSOR}).")
+endif()
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Android")
+    set(OPENSSL_LIB_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../3rdparty/openSSL/static/lib/${OPENSSL_ARCH})
+    set(OPENSSL_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../3rdparty/openSSL/static/include)
+    set(CARGO_CMD OPENSSL_LIB_DIR=${OPENSSL_LIB_DIR} OPENSSL_INCLUDE_DIR=${OPENSSL_INCLUDE_DIR} cargo build)
+else()
+    set(CARGO_CMD cargo build)
+endif()
+
 set(HEADER_FILE vpnglean.h)
 set(LIBNAME ${CMAKE_STATIC_LIBRARY_PREFIX}vpnglean${CMAKE_STATIC_LIBRARY_SUFFIX})
 
 add_custom_target(vpnglean_ffi
     BYPRODUCTS
-        ${CMAKE_CURRENT_BINARY_DIR}/debug/${LIBNAME}
-        ${CMAKE_CURRENT_BINARY_DIR}/release/${LIBNAME}
+        ${CMAKE_CURRENT_BINARY_DIR}/${RUST_ARCH}/debug/${LIBNAME}
+        ${CMAKE_CURRENT_BINARY_DIR}/${RUST_ARCH}/release/${LIBNAME}
         ${CMAKE_CURRENT_BINARY_DIR}/${HEADER_FILE}
+    COMMAND ${CMAKE_COMMAND} -E env rustup target add ${RUST_ARCH}
```

Comment:
We cannot rely on `rustup` to execute successfully at build-time for Linux targets as these must be able to run from an offline environment with networking access specifically disabled. This would need to get moved out of build phase. Perhaps somewhere in `scripts/android/cmake.sh` would be more appropriate?

Reply:
Yes, I mean moving it back there. I actually had forgotten that is where it was from the start 🆘 Thanks Owen <3